### PR TITLE
Added wildcard prefix matching support for ignore list

### PR DESCRIPTION
**Pull request by shrimpy**

### CODE REVIEW

File: KuduSync.NET/KuduSync.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using System.IO.Abstractions;
 using System.Linq;
 using System.Text;
+using System.Text.RegularExpressions;
```
Comment (shrimpy) [inline KuduSync.NET/KuduSync.cs]: removed

File: KuduSync.NET/KuduSync.cs [hunk 2]
```diff
@@ -75,21 +77,31 @@ private bool TryCleanupToBeDeletedDirectory()
...
+                foreach (var token in ingoreTokens)
                 {
-                    throw new NotSupportedException("Wildcard matching (or \\) is not supported");
-                }
+                    var trimedToken = token.Trim();
 
-                return new HashSet<string>(ignoreList, StringComparer.OrdinalIgnoreCase);
+                    if (trimedToken.StartsWith("*"))
+                    {
+                        wildcardIgnoreList.Add(trimedToken.Substring(1));
```
Comment (amitapl) [inline KuduSync.NET/KuduSync.cs]: you can also use TrimStart('*')

File: KuduSync.NET/KuduSync.cs [hunk 3]
```diff
@@ -387,7 +399,8 @@ private DirectoryInfoBase CreateDirectoryInfo(string path)
 
         private bool IgnorePath(FileSystemInfoBase fileSystemInfoBase)
         {
-            return _ignoreList.Contains(fileSystemInfoBase.Name);
+            return (_ignoreList.Contains(fileSystemInfoBase.Name)
+                || _wildcardIgnoreList.Any((name) => fileSystemInfoBase.Name.EndsWith(name, StringComparison.OrdinalIgnoreCase)));
```
Comment (amitapl) [inline KuduSync.NET/KuduSync.cs]: if _wildcardIgnoreList is empty, this is a cheap call right?

**Comment by amitapl** [comment]
:shipit:

File: KuduSync.NET/KuduSync.cs (hunk 3 above)
Comment (shrimpy) [inline KuduSync.NET/KuduSync.cs]: correct. 
and we put "_ignoreList.Contains(fileSystemInfoBase.Name)" first, if we find entry from _ignoreList list. the later condition shouldn`t be executed.

File: KuduSync.NET/KuduSync.cs (hunk 2 above)
Comment (shrimpy) [inline KuduSync.NET/KuduSync.cs]: yep.. updated

**Comment by davidebbo** [comment]
What scenario / user request does this relate to?

**Pull request merged**